### PR TITLE
[CI/OSS] Use new Ubuntu VM scale set agents

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://dev.azure.com/xamarin/public/_build/results?buildId=18233&view=results

Our OSS Linux builds have been failing because they are running out of
disk space.  I've set up a new [virtual machine scale set agent pool][0]
to try to use for our Ubuntu builds.  These VMs are configured with more
disk space than the ~10 GB allowed by the hosted agents.

The virtual machine scale set resource can be found in the azure portal
here:

https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/64e11c84-c922-4ffd-bea9-67ab39354edd/resourceGroups/XAMARIN-PUBLIC-PIPELINES/providers/Microsoft.Compute/virtualMachineScaleSets/android-public-ubuntu-vmss/overview

[0]: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/scale-set-agents?view=azure-devops

### CODE REVIEW

**Comment by pjcollins** [comment]
Latest run was successful - https://dev.azure.com/xamarin/public/_build/results?buildId=18519&view=results.

I think we'll wait for https://github.com/xamarin/xamarin-android/pull/4626#discussion_r424712174 to land first though

**Comment by pjcollins** [comment]
Looks like this is working. I need to figure out how to allow the new vm pool to scale up past two instances though, those settings aren't being retained at the moment for some reason...

**Comment by jonpryor** [comment]
Commit message body:

```
Context: https://dev.azure.com/xamarin/public/_build/results?buildId=18233&view=results

Our OSS Linux builds have been failing beccuse they are running out
of disk space.  I've set up a [virtual machine scale set agent pool][0]
to try to use for our Ubuntu builds.  These VMs are configured with
more disk space than the ~10 GB allowed by the hosted agents.

The virtual machine scale set resource can be found in the
[azure portal][1].

Additionally, move a few variables which controlled job and test
execution behavior into the [pipeline definiton][2].

Finally, a new OSS build specific variable has been added to make sure
we skip a pre-notarization step when creating our OSS `.pkg` installer.

[0]: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/scale-set-agents?view=azure-devops
[1]: https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/64e11c84-c922-4ffd-bea9-67ab39354edd/resourceGroups/XAMARIN-PUBLIC-PIPELINES/providers/Microsoft.Compute/virtualMachineScaleSets/android-public-ubuntu-vmss/overview
[2]: https://dev.azure.com/xamarin/public/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=48&view=Tab_Variables
```

File: build-tools/automation/azure-pipelines-oss.yaml [hunk 1]
```diff
@@ -16,9 +16,6 @@ pr:
 # Global variables
 # Predefined variables: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml
 variables:
```
Comment (mjbond-msft) [inline build-tools/automation/azure-pipelines-oss.yaml:18]: Consider adding a comment that references the build definition variable settings:
https://dev.azure.com/xamarin/public/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=48&view=Tab_Variables

**Comment by pjcollins** [comment]
After further investigation, I've learned that Azure Pipelines automatically manages the scale set instance count based on the demand for the pool. Right now the agent configurations allow for up to 3 VMs at a time, but we should be able to bump this higher as needed in the agent pool settings. This should be good to go now.

**Pull request merged**